app/models/concerns/deposit_state_machine.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Defines the deposit states for an object.
+module DepositStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    # In general, the version status should be used for the status of an object.
+    # However, deposit_state is used for very specific purposes.
+    # In particular, persisting is used to indicate that the DepositWorkJob is queued / running.
+    # Accessioning is used to track if the accessioning of an object originated with H3.
+    # That way when the rabbitMQ message for the end of the accessioning is received,
+    # it can be determined if H3 users should be notified.
+    state_machine :deposit_state, initial: :deposit_none do
+      event :deposit_persist do
+        transition deposit_none: :deposit_persisting
+      end
+
+      # As part of the deposit job, accessioning may or may not be started.
+      # If accessioning is started, the accessioning state is set.
+      # If accessioning is not started, the state is set to none.
+      event :deposit_persist_complete do
+        transition deposit_persisting: :deposit_none
+      end
+
+      event :accession do
+        transition deposit_persisting: :accessioning
+      end
+
+      event :accession_complete do
+        transition accessioning: :deposit_none
+      end
+
+      before_transition accessioning: :deposit_none do |object|
+        Notifier.publish(Notifier::ACCESSIONING_COMPLETE, object:)
+      end
+    end
```

Comment:
This feedback isn't as actionable as I'd like, but one source of confusion in H2 was the nomenclature we used for states and events, and I'm fuzzy on what I'm looking at above. I'd like to see us try to follow two guidelines when naming states and events:

1. Strive for consistency in using nouns / verbs / whatever in naming these (maybe opting to follow the advice here: https://stately.ai/blog/2024-01-23-state-machines-whats-in-a-name#what-or-how-should-i-name-this-thing)
1. Use terms that we developers already use (e.g., I'm not sure I've ever heard a developer talk about "deposits" being "persisted," even though I can make sense of that after a couple seconds of squinting)